client/src/main/java/io/atomix/copycat/client/session/ClientSession.java

```diff
@@ -580,12 +589,30 @@ private ClientSession resetConnection() {
    */
   private ClientSession resetMembers() {
     if (connectMembers.isEmpty() || connectMembers.size() < members.size() - 1) {
-      connectMembers = connectionStrategy.getConnections(leader, new ArrayList<>(members));
+      connectMembers = new ArrayList<>(members);
     }
     return this;
   }
 
   /**
+   * Selects the next member to which to send a request.
+   */
+  private Address selectMember() {
+    // Remove the next member that meets the strategy from the members list.
+    Iterator<Address> iterator = connectMembers.iterator();
+    while (iterator.hasNext()) {
+      Address member = iterator.next();
+      if (connectionStrategy.connect(member, leader)) {
+        iterator.remove();
```

Comment:
Yep. That is what I had in mind as well. The existing `ConnectionStrategy` is more akin to a strategy for selecting a member to connect to. If we can support a strategy for initial session establishment, it would help clients configure what do if they go through all `connectMembers` and still could not establish a session. Retry after a delay is a simple one. We can perhaps toss in exponential back off as you suggested to ensure servers do not get overloaded while they are in the process of bootstrapping the Raft cluster.